e2e/deploy/synthetic.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deploy
+
+import (
+	"errors"
+
+	"github.com/GoogleCloudPlatform/prometheus-engine/e2e/kube"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	SyntheticAppPortName      = "web"
+	SyntheticAppContainerName = "go-synthetic"
+
+	appManifest = "../examples/instrumentation/go-synthetic/go-synthetic.yaml"
```

Comment:
We won't have this in path in google3 - it only exists in this relative location in Github, so we won't be able to use these setup methods there unless we wanted to start maintaining a mirror internally and move everything over periodically...